qutip/qip/qasm.py

```diff
@@ -0,0 +1,735 @@
+            if gate_defn_mode:
+                if command[0] == "{":
+                    gate_defn_mode = False
+                    open_bracket_mode = True
+                    gate_elems = []
+                    continue
+                else:
+                    raise SyntaxError("QASM: incorrect bracket formatting")
+            elif open_bracket_mode:
+                if command[0] == "{":
+                    raise SyntaxError("QASM: incorrect bracket formatting")
+                elif command[0] == "}":
+                    if not curr_gate.gates_inside:
+                        raise NotImplementedError("QASM: opaque gate {} are  \
+                                                   not allowed, please define \
+                                                   or omit \
+                                                   them".format(curr_gate.name))
+                    open_bracket_mode = False
+                    self.gate_names.add(curr_gate.name)
+                    self.qasm_gates[curr_gate.name] = curr_gate
+                    continue
+                elif command[0] in self.gate_names:
+                    name = command[0]
+                    gate_args, gate_regs = _gate_processor(command)
+                    gate_added = self.qasm_gates[name]
+                    curr_gate.gates_inside.append([name,
+                                                   gate_args,
+                                                   gate_regs])
+            elif command[0] == "gate":
+                gate_name = command[1]
+                gate_args, gate_regs = _gate_processor(command[1:])
+                curr_gate = QasmGate(gate_name, gate_args, gate_regs)
+                gate_defn_mode = True
+
+            elif command[0] == "qreg":
+                groups = re.match(r"(.*)\[(.*)\]", "".join(command[1:]))
+                if groups:
+                    qubit_name = groups.group(1)
+                    num_regs = int(groups.group(2))
+                    self.qubit_regs[qubit_name] = list(range(
+                                                self.num_qubits,
+                                                self.num_qubits + num_regs))
+                    self.num_qubits += num_regs
+                else:
+                    raise SyntaxError("QASM: incorrect bracket formatting")
+
+            elif command[0] == "creg":
+                groups = re.match(r"(.*)\[(.*)\]", "".join(command[1:]))
+                if groups:
+                    cbit_name = groups.group(1)
+                    num_regs = int(groups.group(2))
+                    self.cbit_regs[cbit_name] = list(range(self.num_cbits,
+                                                     self.num_cbits + num_regs))
+                    self.num_cbits += num_regs
+                else:
+                    raise SyntaxError("QASM: incorrect bracket formatting")
+            elif command[0] in ["barrier", "include", "reset"]:
```

Comment:
Yep, that is what reset is meant to do? I did not think to accommodate it since it seems like a scarcely used feature and did not merit a separate kind of processing with QuTiP. What do you think? Maybe a warning is appropriate here? Or an error?

Reply:
If `reset` is not supported, i.e. qasm circuit with `reset` will be wrong in QuTiP, it deserves a `NotImplementedError`.